### PR TITLE
[FXML-4416] Add EmitC lowering for arith.cmpi

### PR DESCRIPTION
**Pull request by cferry-AMD**
Piggybacking on @mgehre-amd 's work, this PR implements arith.cmpi in the ArithToEmitC pass. This conversion currently only supports integers.

### CODE REVIEW

File: mlir/lib/Conversion/ArithToEmitC/ArithToEmitC.cpp [hunk 1]
```diff
@@ -160,6 +160,78 @@ class CmpFOpConversion : public OpConversionPattern<arith::CmpFOp> {
...
+      return emitc::CmpPredicate::le;
+    case arith::CmpIPredicate::sgt:
+    case arith::CmpIPredicate::ugt:
+      return emitc::CmpPredicate::gt;
+    case arith::CmpIPredicate::sge:
+    case arith::CmpIPredicate::uge:
+      return emitc::CmpPredicate::ge;
+    }
+    llvm_unreachable("unknown cmpi predicate kind");
+  }
```
Comment (mgehre-amd) [inline mlir/lib/Conversion/ArithToEmitC/ArithToEmitC.cpp:205]: nit: please add a newline after this line

File: mlir/test/Conversion/ArithToEmitC/arith-to-emitc.mlir [hunk 2]
```diff
@@ -173,3 +173,25 @@ func.func @arith_int_to_float_cast_ops(%arg0: i8, %arg1: i64) {
 
   return
 }
+
+// -----
+
```
Comment (mgehre-amd) [inline mlir/test/Conversion/ArithToEmitC/arith-to-emitc.mlir:178]: We did exhaustive testing for the cmpf predicates, so I guess we should do it here, too.
The interesting thing about the remaining operators (after eq and ult) for me is
1) that they use the right predicate in emitc
2) that they use the signed/unsigned.

So I think you can add a test case after this two extensive ones,
 that just checks those two properties (and not everything) like
```
func.func @arith_cmpi(%arg0: i32, %arg1: i32) -> i1 {
 %ule = arith.cmpi ule, %arg0, %arg1 : i32
  // CHECK: emitc.cmp le, {{.*}} : (ui32, ui32) -> i1
 %sle = arith.cmpi sle, %arg0, %arg1 : i32
  // CHECK: emitc.cmp le, {{.*}} : (i32, i32) -> i1
 %ne = arith.cmpi ne, %arg0, %arg1 : i32
  // CHECK: emitc.cmp ne, {{.*}} : (i32, i32) -> i1
...
}

```

File: mlir/test/Conversion/ArithToEmitC/arith-to-emitc.mlir [hunk 3]
```diff
@@ -173,3 +173,25 @@ func.func @arith_int_to_float_cast_ops(%arg0: i8, %arg1: i64) {
...
+func.func @arith_cmpi_ult(%arg0: i32, %arg1: i32) -> i1 {
+  // CHECK-LABEL: arith_cmpi_ult
+  // CHECK-SAME: ([[Arg0:[^ ]*]]: i32, [[Arg1:[^ ]*]]: i32)
+  // CHECK-DAG: [[CastArg0:[^ ]*]] = emitc.cast [[Arg0]] : i32 to ui32
+  // CHECK-DAG: [[CastArg1:[^ ]*]] = emitc.cast [[Arg1]] : i32 to ui32
+  // CHECK-DAG: [[ULT:[^ ]*]] = emitc.cmp lt, [[CastArg0]], [[CastArg1]] : (ui32, ui32) -> i1
+  %eq = arith.cmpi ult, %arg0, %arg1 : i32
+  // CHECK: return [[ULT]]
+  return %eq: i1
```
Comment (mgehre-amd) [inline mlir/test/Conversion/ArithToEmitC/arith-to-emitc.mlir:196]: nit:
```suggestion
  %ult = arith.cmpi ult, %arg0, %arg1 : i32
  // CHECK: return [[ULT]]
  return %ult: i1
```

File: mlir/test/Conversion/ArithToEmitC/arith-to-emitc.mlir (hunk 2 above)
Comment (cferry-AMD) [reply mlir/test/Conversion/ArithToEmitC/arith-to-emitc.mlir:178]: Done, I guess :)

**Review by mgehre-amd — APPROVED** [review]
Thanks!

**Review by TinaAMD — APPROVED** [review]
LGTM

**Pull request merged**